### PR TITLE
ESLint Plugin/Scripts: Update ESLint and related deps to 7.1.0

### LINKED ISSUE #20235: .eslintrc.js linked to @wordpress/scripts may cause "The "path" argument must be of type string. Received type undefined" error
**Describe the bug**
To ensure VSCode is using the same ESLint config as `wp-scripts lint-js` I created a file `.eslintrc.js` with

```js
const defaultConfig = require('./node_modules/@wordpress/scripts/config/.eslintrc.js' );
module.exports = defaultConfig;
```

This is working fine unless the project is a sub-project.
image

**To reproduce**
1. Create a directory `project`
1. Inside that directory create a directory named `sub-project`
1. In `sub-project` create `package.json` with
   ```json
   {
     "name": "test",
     "devDependencies": {
       "@wordpress/scripts": "^7.1.0"
     }
   }
   ```
1. In `sub-project` create `.eslintrc.js` with
   ```js
   const defaultConfig = require('./node_modules/@wordpress/scripts/config/.eslintrc.js' );
   module.exports = defaultConfig;
   ```
1. Run `npm i`
1. Open  the directory `project` in VSCode
1. In `sub-project` create a file `test.js` with `console.log( "Hello World" );`
1. Notice that no linting error is shown.

In the console of ESLint you should also see this line:

```
[Info  - 11:43:54] Cannot read config file: /Users/Dominik/Desktop/project/sub-project/.eslintrc.js Error: The "path" argument must be of type string. Received type undefined
```

I tracked this down to https://github.com/WordPress/gutenberg/blob/95be29428a6ee23649e46136323c5f0f4c6b540a/packages/scripts/utils/file.js#L13 where `getPackagePath()` is returning undefined.

If `sub-project` is directly opened in VS Code the path returned by `getPackagePath()` is  `/Users/Dominik/Desktop/project/sub-project/package.json`

**Additional context**

<details>
<summary>Debug log of VS Code</summary>

```
[Info  - 11:31:55] ESLint server stopped.
[Info  - 11:31:56] ESLint server running in node v12.4.0
[Info  - 11:31:56] ESLint server is running.
[Info  - 11:31:56] ESLint library loaded from: /Users/Dominik/Desktop/project/sub-project/node_modules/eslint/lib/api.js
2020-02-14T10:31:56.650Z eslint:ignore-pattern Create with: [ IgnorePattern { patterns: [ '/node_modules/*', '/bower_components/*' ], basePath: '/Users/Dominik/Desktop/project', loose: false } ]
2020-02-14T10:31:56.652Z eslint:ignore-pattern   processed: { basePath: '/Users/Dominik/Desktop/project', patterns: [ '/node_modules/*', '/bower_components/*' ] }
2020-02-14T10:31:56.653Z eslint:ignore-pattern Create with: [ IgnorePattern { patterns: [ '/node_modules/*', '/bower_components/*' ], basePath: '/Users/Dominik/Desktop/project', loose: false } ]
2020-02-14T10:31:56.653Z eslint:ignore-pattern   processed: { basePath: '/Users/Dominik/Desktop/project', patterns: [ '/node_modules/*', '/bower_components/*' ] }
2020-02-14T10:31:56.653Z eslint:cascading-config-array-factory Load config files for /Users/Dominik/Desktop/project/sub-project/node_modules/@wordpress/scripts/utils.
2020-02-14T10:31:56.653Z eslint:cascading-config-array-factory No cache found: /Users/Dominik/Desktop/project/sub-project/node_modules/@wordpress/scripts/utils.
2020-02-14T10:31:56.654Z eslint:config-array-factory Config file not found on /Users/Dominik/Desktop/project/sub-project/node_modules/@wordpress/scripts/utils
2020-02-14T10:31:56.654Z eslint:cascading-config-array-factory No cache found: /Users/Dominik/Desktop/project/sub-project/node_modules/@wordpress/scripts.
2020-02-14T10:31:56.654Z eslint:config-array-factory Loading package.json config file: /Users/Dominik/Desktop/project/sub-project/node_modules/@wordpress/scripts/package.json
2020-02-14T10:31:56.654Z eslint:config-array-factory Loading JSON config file: /Users/Dominik/Desktop/project/sub-project/node_modules/@wordpress/scripts/package.json
2020-02-14T10:31:56.655Z eslint:config-array-factory Error reading package.json file: /Users/Dominik/Desktop/project/sub-project/node_modules/@wordpress/scripts/package.json
2020-02-14T10:31:56.655Z eslint:config-array-factory Config file not found on /Users/Dominik/Desktop/project/sub-project/node_modules/@wordpress/scripts
2020-02-14T10:31:56.655Z eslint:cascading-config-array-factory No cache found: /Users/Dominik/Desktop/project/sub-project/node_modules/@wordpress.
2020-02-14T10:31:56.655Z eslint:config-array-factory Config file not found on /Users/Dominik/Desktop/project/sub-project/node_modules/@wordpress
2020-02-14T10:31:56.655Z eslint:cascading-config-array-factory No cache found: /Users/Dominik/Desktop/project/sub-project/node_modules.
2020-02-14T10:31:56.655Z eslint:config-array-factory Config file not found on /Users/Dominik/Desktop/project/sub-project/node_modules
2020-02-14T10:31:56.655Z eslint:cascading-config-array-factory No cache found: /Users/Dominik/Desktop/project/sub-project.
2020-02-14T10:31:56.656Z eslint:config-array-factory Loading JS config file: /Users/Dominik/Desktop/project/sub-project/.eslintrc.js
getPackagePath:undefined <--- The error, console.log for getPackagePath() added by myself
2020-02-14T10:31:56.793Z eslint:config-array-factory Error reading JavaScript file: /Users/Dominik/Desktop/project/sub-project/.eslintrc.js
(node:52624) UnhandledPromiseRejectionWarning: TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type undefined
    at validateString (internal/validators.js:107:11)
    at Object.dirname (path.js:1121:5)
    at fromProjectRoot (/Users/Dominik/Desktop/project/sub-project/node_modules/@wordpress/scripts/utils/file.js:13:18)
    at hasProjectFile (/Users/Dominik/Desktop/project/sub-project/node_modules/@wordpress/scripts/utils/file.js:16:14)
    at hasPrettierConfig (/Users/Dominik/Desktop/project/sub-project/node_modules/@wordpress/scripts/utils/config.js:35:2)
    at Object.<anonymous> (/Users/Dominik/Desktop/project/sub-project/node_modules/@wordpress/scripts/config/.eslintrc.js:20:8)
    at Module._compile (internal/modules/cjs/loader.js:786:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:798:10)
    at Module.load (internal/modules/cjs/loader.js:645:32)
    at Function.Module._load (internal/modules/cjs/loader.js:560:12)
(node:52624) UnhandledPromiseRejectionWarning: TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type undefined
    at validateString (internal/validators.js:107:11)
    at Object.dirname (path.js:1121:5)
    at fromProjectRoot (/Users/Dominik/Desktop/project/sub-project/node_modules/@wordpress/scripts/utils/file.js:13:18)
    at hasProjectFile (/Users/Dominik/Desktop/project/sub-project/node_modules/@wordpress/scripts/utils/file.js:16:14)
    at hasPrettierConfig (/Users/Dominik/Desktop/project/sub-project/node_modules/@wordpress/scripts/utils/config.js:35:2)
    at Object.<anonymous> (/Users/Dominik/Desktop/project/sub-project/node_modules/@wordpress/scripts/config/.eslintrc.js:20:8)
    at Module._compile (internal/modules/cjs/loader.js:786:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:798:10)
    at Module.load (internal/modules/cjs/loader.js:645:32)
    at Function.Module._load (internal/modules/cjs/loader.js:560:12)
(node:52624) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:52624) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:52624) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
(node:52624) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
2020-02-14T10:31:56.795Z eslint:ignore-pattern Create with: [ IgnorePattern { patterns: [ '/node_modules/*', '/bower_components/*' ], basePath: '/Users/Dominik/Desktop/project', loose: false } ]
2020-02-14T10:31:56.795Z eslint:ignore-pattern   processed: { basePath: '/Users/Dominik/Desktop/project', patterns: [ '/node_modules/*', '/bower_components/*' ] }
2020-02-14T10:31:56.795Z eslint:ignore-pattern Create with: [ IgnorePattern { patterns: [ '/node_modules/*', '/bower_components/*' ], basePath: '/Users/Dominik/Desktop/project', loose: false } ]
2020-02-14T10:31:56.795Z eslint:ignore-pattern   processed: { basePath: '/Users/Dominik/Desktop/project', patterns: [ '/node_modules/*', '/bower_components/*' ] }
2020-02-14T10:31:56.796Z eslint:cascading-config-array-factory Load config files for /Users/Dominik/Desktop/project/sub-project/node_modules/@wordpress/scripts/utils.
2020-02-14T10:31:56.796Z eslint:cascading-config-array-factory No cache found: /Users/Dominik/Desktop/project/sub-project/node_modules/@wordpress/scripts/utils.
2020-02-14T10:31:56.796Z eslint:config-array-factory Config file not found on /Users/Dominik/Desktop/project/sub-project/node_modules/@wordpress/scripts/utils
2020-02-14T10:31:56.796Z eslint:cascading-config-array-factory No cache found: /Users/Dominik/Desktop/project/sub-project/node_modules/@wordpress/scripts.
2020-02-14T10:31:56.796Z eslint:config-array-factory Loading package.json config file: /Users/Dominik/Desktop/project/sub-project/node_modules/@wordpress/scripts/package.json
2020-02-14T10:31:56.796Z eslint:config-array-factory Loading JSON config file: /Users/Dominik/Desktop/project/sub-project/node_modules/@wordpress/scripts/package.json
2020-02-14T10:31:56.797Z eslint:config-array-factory Error reading package.json file: /Users/Dominik/Desktop/project/sub-project/node_modules/@wordpress/scripts/package.json
2020-02-14T10:31:56.797Z eslint:config-array-factory Config file not found on /Users/Dominik/Desktop/project/sub-project/node_modules/@wordpress/scripts
2020-02-14T10:31:56.797Z eslint:cascading-config-array-factory No cache found: /Users/Dominik/Desktop/project/sub-project/node_modules/@wordpress.
2020-02-14T10:31:56.797Z eslint:config-array-factory Config file not found on /Users/Dominik/Desktop/project/sub-project/node_modules/@wordpress
2020-02-14T10:31:56.797Z eslint:cascading-config-array-factory No cache found: /Users/Dominik/Desktop/project/sub-project/node_modules.
2020-02-14T10:31:56.797Z eslint:config-array-factory Config file not found on /Users/Dominik/Desktop/project/sub-project/node_modules
2020-02-14T10:31:56.797Z eslint:cascading-config-array-factory No cache found: /Users/Dominik/Desktop/project/sub-project.
2020-02-14T10:31:56.797Z eslint:config-array-factory Loading JS config file: /Users/Dominik/Desktop/project/sub-project/.eslintrc.js
2020-02-14T10:31:56.798Z eslint:config-array-factory Error reading JavaScript file: /Users/Dominik/Desktop/project/sub-project/.eslintrc.js
[Info  - 11:31:56] Cannot read config file: /Users/Dominik/Desktop/project/sub-project/.eslintrc.js Error: The "path" argument must be of type string. Received type undefined
2020-02-14T10:43:54.939Z eslint:ignore-pattern Create with: [ IgnorePattern { patterns: [ '/node_modules/*', '/bower_components/*' ], basePath: '/Users/Dominik/Desktop/project', loose: false } ]
2020-02-14T10:43:54.939Z eslint:ignore-pattern   processed: { basePath: '/Users/Dominik/Desktop/project', patterns: [ '/node_modules/*', '/bower_components/*' ] }
2020-02-14T10:43:54.939Z eslint:ignore-pattern Create with: [ IgnorePattern { patterns: [ '/node_modules/*', '/bower_components/*' ], basePath: '/Users/Dominik/Desktop/project', loose: false } ]
2020-02-14T10:43:54.939Z eslint:ignore-pattern   processed: { basePath: '/Users/Dominik/Desktop/project', patterns: [ '/node_modules/*', '/bower_components/*' ] }
2020-02-14T10:43:54.939Z eslint:cascading-config-array-factory Load config files for /Users/Dominik/Desktop/project/sub-project.
2020-02-14T10:43:54.939Z eslint:cascading-config-array-factory No cache found: /Users/Dominik/Desktop/project/sub-project.
2020-02-14T10:43:54.940Z eslint:config-array-factory Loading JS config file: /Users/Dominik/Desktop/project/sub-project/.eslintrc.js
2020-02-14T10:43:54.940Z eslint:config-array-factory Error reading JavaScript file: /Users/Dominik/Desktop/project/sub-project/.eslintrc.js
(node:52624) UnhandledPromiseRejectionWarning: TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type undefined
    at validateString (internal/validators.js:107:11)
    at Object.dirname (path.js:1121:5)
    at fromProjectRoot (/Users/Dominik/Desktop/project/sub-project/node_modules/@wordpress/scripts/utils/file.js:13:18)
    at hasProjectFile (/Users/Dominik/Desktop/project/sub-project/node_modules/@wordpress/scripts/utils/file.js:16:14)
    at hasPrettierConfig (/Users/Dominik/Desktop/project/sub-project/node_modules/@wordpress/scripts/utils/config.js:35:2)
    at Object.<anonymous> (/Users/Dominik/Desktop/project/sub-project/node_modules/@wordpress/scripts/config/.eslintrc.js:20:8)
    at Module._compile (internal/modules/cjs/loader.js:786:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:798:10)
    at Module.load (internal/modules/cjs/loader.js:645:32)
    at Function.Module._load (internal/modules/cjs/loader.js:560:12)
(node:52624) UnhandledPromiseRejectionWarning: TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type undefined
    at validateString (internal/validators.js:107:11)
    at Object.dirname (path.js:1121:5)
    at fromProjectRoot (/Users/Dominik/Desktop/project/sub-project/node_modules/@wordpress/scripts/utils/file.js:13:18)
    at hasProjectFile (/Users/Dominik/Desktop/project/sub-project/node_modules/@wordpress/scripts/utils/file.js:16:14)
    at hasPrettierConfig (/Users/Dominik/Desktop/project/sub-project/node_modules/@wordpress/scripts/utils/config.js:35:2)
    at Object.<anonymous> (/Users/Dominik/Desktop/project/sub-project/node_modules/@wordpress/scripts/config/.eslintrc.js:20:8)
    at Module._compile (internal/modules/cjs/loader.js:786:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:798:10)
    at Module.load (internal/modules/cjs/loader.js:645:32)
    at Function.Module._load (internal/modules/cjs/loader.js:560:12)
(node:52624) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
(node:52624) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
2020-02-14T10:43:54.941Z eslint:ignore-pattern Create with: [ IgnorePattern { patterns: [ '/node_modules/*', '/bower_components/*' ], basePath: '/Users/Dominik/Desktop/project', loose: false } ]
2020-02-14T10:43:54.941Z eslint:ignore-pattern   processed: { basePath: '/Users/Dominik/Desktop/project', patterns: [ '/node_modules/*', '/bower_components/*' ] }
2020-02-14T10:43:54.941Z eslint:ignore-pattern Create with: [ IgnorePattern { patterns: [ '/node_modules/*', '/bower_components/*' ], basePath: '/Users/Dominik/Desktop/project', loose: false } ]
2020-02-14T10:43:54.942Z eslint:ignore-pattern   processed: { basePath: '/Users/Dominik/Desktop/project', patterns: [ '/node_modules/*', '/bower_components/*' ] }
2020-02-14T10:43:54.942Z eslint:cascading-config-array-factory Load config files for /Users/Dominik/Desktop/project/sub-project.
2020-02-14T10:43:54.942Z eslint:cascading-config-array-factory No cache found: /Users/Dominik/Desktop/project/sub-project.
2020-02-14T10:43:54.942Z eslint:config-array-factory Loading JS config file: /Users/Dominik/Desktop/project/sub-project/.eslintrc.js
2020-02-14T10:43:54.942Z eslint:config-array-factory Error reading JavaScript file: /Users/Dominik/Desktop/project/sub-project/.eslintrc.js
[Info  - 11:43:54] Cannot read config file: /Users/Dominik/Desktop/project/sub-project/.eslintrc.js Error: The "path" argument must be of type string. Received type undefined
```

</details>

### LINKED ISSUE #22385: Scripts: Upgrade ESLint to the new major version (7.x)
## Description

ESLint `v7.0.0` was released:

https://eslint.org/blog/2020/05/eslint-v7.0.0-released

We should update our packages to use the latest version. It has a few breaking changes as highlighted here:
https://eslint.org/blog/2020/05/eslint-v7.0.0-released#breaking-changes

One of the issues resolved (https://github.com/eslint/eslint/issues/11558) allows us to remove the following workaround added in https://github.com/WordPress/gutenberg/pull/17744:
https://github.com/WordPress/gutenberg/blob/605cb2bc2fd5906cec72c9ae357f5b3c6f0e00c5/packages/scripts/config/.eslintrc.js#L6-L10

## References

It might be helpful to look at previous PR where minor version upgrade for ESLint was applied: https://github.com/WordPress/gutenberg/pull/21424. The changes included:
- upgrade of ESLint package to the latest version
- bringing all ESLint plugins and configs used to the latest version
- listing all changes applied in the relevant changelog files
- fixing any new errors that might pop up

### PR DESCRIPTION
**Pull request by ocean90**
## Description

Fixes #22385.
Fixes #20235.

This is a first pass for updating @wordpress/eslint-plugin and @wordpress/scripts to ESLint 7.x.

What I have done:

```
# Remove eslint from packages/scripts/package.json
npm i
npx lerna add eslint packages/scripts
# Remove eslint-plugin-react-hooks from packages/eslint-plugin/package.json
npm i
npx lerna add eslint-plugin-react-hooks packages/eslint-plugin
# Remove eslint-plugin-jsdoc from packages/eslint-plugin/package.json
npm i
npx lerna add eslint-plugin-jsdoc packages/eslint-plugin
# Remove eslint-plugin-react from packages/eslint-plugin/package.json
npm i
npx lerna add eslint-plugin-react packages/eslint-plugin
```

Changelogs:

* eslint-plugin-react-hooks: https://github.com/facebook/react/blob/master/packages/eslint-plugin-react-hooks/CHANGELOG.md
* eslint-plugin-jsdoc: https://github.com/gajus/eslint-plugin-jsdoc/releases ([Commits](https://github.com/gajus/eslint-plugin-jsdoc/compare/v22.1.0...v26.0.0))
* eslint-plugin-react: https://github.com/yannickcr/eslint-plugin-react/blob/master/CHANGELOG.md

There two remaining deps which haven't got an update yet:

* https://github.com/benmosher/eslint-plugin-import/issues/1772
* https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/issues/683

---

~During the commit I got a package.lock failure:~

<details>
<summary>Error Details</summary>

```
> node ./bin/validate-package-lock.js "/Users/Dominik/Development/WordPress/wordpress-core/src/wp-content/plugins/gutenberg/package-lock.json"

Invalid resolved dependency in package-lock.json.

{
	"node-pre-gyp": {
		"version": "0.12.0",
		"resolved": false,
		"integrity": "sha512-4KghwV8vH5k+g2ylT+sLTjy5wmUOb9vPhnM8NHvRf9dHmnW/CndrFXy2aRPaPST6dugXSdHXfeaHQm77PIz/1A==",
		"dev": true,
		"optional": true,
		"requires": {
			"detect-libc": "^1.0.2",
			"mkdirp": "^0.5.1",
			"needle": "^2.2.1",
			"nopt": "^4.0.1",
			"npm-packlist": "^1.1.6",
			"npmlog": "^4.0.2",
			"rc": "^1.2.7",
			"rimraf": "^2.6.1",
			"semver": "^5.3.0",
			"tar": "^4"
		},
		"dependencies": {
			"rimraf": {
				"version": "2.7.1",
				"resolved": "https://registry.npmjs.org/rimraf/-/rimraf-2.7.1.tgz",
				"integrity": "sha512-uWjbaKIK3T1OSVptzX7Nl6PvQ3qAGtKEtVRjRuazjfL3Bx5eI409VZSqgND+4UNnmzLVdPj9FqFJNPqBZFve4w==",
				"dev": true,
				"optional": true,
				"requires": {
					"glob": "^7.1.3"
				}
			}
		}
	}
}
```

</details>

~I tried removing node_modules but that didn't solve it. What else can I try? I'm using npm 6.14.5.~ Manually fixed in https://github.com/WordPress/gutenberg/pull/22771/commits/33b287bdbae47aa624370ee87271730b085ecb3d

## How has this been tested?
`wp-scripts lint-js` which produced a lot of `Missing JSDoc` and `Missing @param` errors (see on [Travis](https://travis-ci.com/github/WordPress/gutenberg/jobs/341713503#L436)). Probably due to changes in eslint-plugin-jsdoc v25.0.0 https://github.com/gajus/eslint-plugin-jsdoc/releases/tag/v25.0.0.

## Types of changes
Breaking change

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by ocean90** [comment]
The JSDoc lint errors seem to be correct and should probably get fixed in a separate PR before the ESLint update gets merged?

Some examples:

```
[1] /home/travis/build/WordPress/gutenberg/packages/annotations/src/store/actions.js
[1]    6:1  error  Missing JSDoc @param "annotation.selector" declaration  jsdoc/require-param
[1]    6:1  error  Missing JSDoc @param "annotation.source" declaration    jsdoc/require-param
[1]    6:1  error  Missing JSDoc @param "annotation.id" declaration        jsdoc/require-param
[1]   16:0  error  Missing @param "annotation.selector"                    jsdoc/check-param-names
[1]   16:0  error  Missing @param "annotation.source"                      jsdoc/check-param-names
[1]   16:0  error  Missing @param "annotation.id"                          jsdoc/check-param-names
```

https://github.com/WordPress/gutenberg/blob/78587c9ae30be88cf3414ae66d2f5883f3b18623/packages/annotations/src/store/actions.js#L22-L24

Syntax for optional props changed?

---

```
[1] /home/travis/build/WordPress/gutenberg/packages/annotations/src/store/actions.js
[1]    6:1  error  Missing JSDoc @param "annotation.selector" declaration  jsdoc/require-param
[1]    6:1  error  Missing JSDoc @param "annotation.source" declaration    jsdoc/require-param
[1]    6:1  error  Missing JSDoc @param "annotation.id" declaration        jsdoc/require-param
[1]   16:0  error  Missing @param "annotation.selector"                    jsdoc/check-param-names
[1]   16:0  error  Missing @param "annotation.source"                      jsdoc/check-param-names
[1]   16:0  error  Missing @param "annotation.id"                          jsdoc/check-param-names
```

https://github.com/WordPress/gutenberg/blob/78587c9ae30be88cf3414ae66d2f5883f3b18623/packages/block-directory/src/store/actions.js#L53-L60

The documentation is missing.

---

```
[1] /home/travis/build/WordPress/gutenberg/packages/block-editor/src/autocompleters/block.js
[1]   84:1  error  Missing JSDoc @param "root0" declaration                                  jsdoc/require-param
[1]   84:1  error  Missing JSDoc @param "root0.getBlockInsertionParentClientId" declaration  jsdoc/require-param
[1]   84:1  error  Missing JSDoc @param "root0.getInserterItems" declaration                 jsdoc/require-param
[1]   84:1  error  Missing JSDoc @param "root0.getSelectedBlockName" declaration             jsdoc/require-param
```

https://github.com/WordPress/gutenberg/blob/78587c9ae30be88cf3414ae66d2f5883f3b18623/packages/block-editor/src/autocompleters/block.js#L84-L94

The documentation is missing.

**Comment by gziolo** [comment]
> The JSDoc lint errors seem to be correct and should probably get fixed in a separate PR before the ESLint update gets merged?

It looks like there are 2 rules that are more strict now based on failures from:
https://travis-ci.com/github/WordPress/gutenberg/jobs/341716759

Those rules are
- `jsdoc/require-param`
- `jsdoc/check-param-name`

I agree that it makes sense to address them separately. On the technical side of things, it might be simpler to downgrade them to warnings and bring back to errors in two follow up PRs for each rule separately.

I see also that some unit tests are failing:
https://travis-ci.com/github/WordPress/gutenberg/jobs/341716764

This is very surprising but it looks like those are valid issues since the impact ESLint related logic:
- `packages/eslint-plugin/rules/__tests__/gutenberg-phase.js`
- `packages/eslint-plugin/rules/__tests__/no-unused-vars-before-return.js`


> There two remaining deps which haven't got an update yet:
> 
> - benmosher/eslint-plugin-import#1772
> - jsx-eslint/eslint-plugin-jsx-a11y#683

Maybe, they don't need to be updated. I think it's quite common that those plugins release at their own pace.

File: packages/eslint-plugin/CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,9 @@
 ### Breaking Changes
 
 - The bundled `wp-prettier` dependency has been upgraded from `1.19.1` to `2.0.5`. Refer to the [Prettier 2.0 "2020" blog post](https://prettier.io/blog/2020/03/21/2.0.0.html) for full details about the major changes included in Prettier 2.0.
+- The bundled `eslint-plugin-react-hooks` dependency has been updated from requiring `^3.0.0` to requiring `^4.0.4`.
+- The bundled `eslint-plugin-jsdoc` dependency has been updated from requiring `^22.1.0` to requiring `^26.0.0`.
+- The bundled `eslint-plugin-react` dependency has been updated from requiring `^7.19.0` to requiring `^7.20.0`.
```
Comment (gziolo) [inline packages/eslint-plugin/CHANGELOG.md:10]: `eslint-plugin-react` – this one should be moved to the `Enhancements` section. It looks like a minor update.

**Comment by ocean90** [comment]
@gziolo Thanks for the feedback!

Nice catch with the failing unit tests, looks like I focused to match on the lint errors. Fixed in  https://github.com/WordPress/gutenberg/pull/22771/commits/338639acd60dce355308954df930150ddb62fc74.

> I agree that it makes sense to address them separately. On the technical side of things, it might be simpler to downgrade them to warnings and bring back to errors in two follow up PRs for each rule separately.

I'm not sure, might be easier to just fix them in one in one sweep. The error report might look severe but the `jsdoc/require-param`/`jsdoc/check-param-name` are just duplicated for the same params.

> Maybe, they don't need to be updated.

Yes, I think we can release without them for now, at least there seem to be no issues. It's just about the npm warning:

```
npm WARN eslint-plugin-jsx-a11y@6.2.3 requires a peer of eslint@^3 || ^4 || ^5 || ^6 but none is installed. You must install peer dependencies yourself.
```

**Comment by ocean90** [comment]
@gziolo Okay, after fixing some issues I think we should go with your downgrade to warning idea here (https://github.com/WordPress/gutenberg/pull/22771/commits/5757cc950e45ad23d4eb3023a8466c72b6f2cc0b) which should make it easier to fix the errors in follow-up PRs. Switching branches isn't so efficient and only relying on the current Travis output isn't good either because there may be follow-up errors like shown in #22794.

Marking this PR as ready for review and will continue with a tracking issue for the JSDoc update if you all agree with the proposed workflow.

**Comment by gziolo** [comment]
There is one more option for how we can approach it. Now that this branch passes, we can open another branch targeting this branch that fixes individual rules (or both together if they are tightly coupled). Otherwise, we would have to update `@wordpress/eslint-plugin` package rather than `.eslintrc.js` which hides the issue for Gutenberg only but not for 3rd party projects.

**Comment by ocean90** [comment]
> Otherwise, we would have to update `@wordpress/eslint-plugin` package rather than `.eslintrc.js` which hides the issue for Gutenberg only but not for 3rd party projects.

It hides them only temporarily for the project until we have fixed the issues. 3rd party projects should follow the same procedure or fix them directly without the warning step since those are valid errors. So I don't think we should change the severity in the published package.

**Comment by ocean90** [comment]
Related chat on [Slack](https://make.wordpress.org/chat/): https://wordpress.slack.com/archives/C5UNMSU4R/p1591107916367900

The decision was to merge this PR as is and then I'll follow-up with a tracking ticket for the new JSDoc warnings.

**Review by gziolo — APPROVED** [review]
It works as expected, there are new warnings introduced:

```
✖ 215 problems (0 errors, 215 warnings)
  0 errors and 129 warnings potentially fixable with the `--fix` option.
```

Let's move forward according to the plan discussed earlier this week during the Core JS meeting 🚢

**Pull request merged**